### PR TITLE
layers: Add SPIR-V Instruction Table

### PR DESCRIPTION
**Pull request by sfricke-samsung**
`spirv-val` has this really nice function for getting basic information about SPIR-V instructions

```cpp
uint32_t ValidationState_t::GetTypeId(uint32_t id) const {
  const Instruction* inst = FindDef(id);
  return inst ? inst->type_id() : 0;
}
```

and what I realize is there is lacking information about each SPIR-V opcode in the VVL to get information such as `inst->type_id()`

SPIRV-Tools also [generates](https://github.com/KhronosGroup/SPIRV-Tools/blob/8a40f6de57d7b78bc431678d90aa8a570d1631f2/utils/generate_grammar_tables.py#L309) all there information and creates a [table](https://github.com/KhronosGroup/SPIRV-Tools/blob/8a40f6de57d7b78bc431678d90aa8a570d1631f2/source/table.h#L22-L45) to do hashmap look ups.

since we already now have the infrastructure in place, it was easy to quick add actually :smile:

### CODE REVIEW

File: scripts/spirv_grammar_generator.py [hunk 1]
```diff
@@ -245,20 +251,56 @@ def parseGrammar(self, grammar):
...
+            output += '    uint32_t image_operands_position; // operand ID position or zero if not present\n'
+            output += '};\n'
+            output += '\n'
+            output += '// Static table to replace having many large switch statement functions for looking up each part\n'
+            output += '// of a given SPIR-V opcode instruction\n'
+            output += '//\n'
+            output += '// clang-format off\n'
+            output += 'static const layer_data::unordered_map<uint32_t, InstructionInfo> kInstructionTable {\n'
+            for opcode, info in sorted(self.opcodes.items()):
+                output += '    {{spv::{}, {{\"{}\", {}, {}, {}, {}, {}}}}},\n'.format(
```
Comment (ncesario-lunarg) [inline scripts/spirv_grammar_generator.py:292]: I personally find `f'{{spv::{info["name"]}, {{"{info["name"]}", ...'` a lot easier to read, but I know I'm in the minority over here ;)

**Review by ncesario-lunarg — APPROVED** [review]
LGTM! Only problem is the following we're seeing in CI:
```
[ RUN      ] VkLayerTest.DeviceMemoryScope
C:\j\w\32-bit\Release\Vulkan-ValidationLayers\tests\vkrenderframework.cpp(196): error: Failed
Did not receive expected error 'VUID-RuntimeSpirv-vulkanMemoryModel-06265'
[  FAILED  ] VkLayerTest.DeviceMemoryScope (602 ms)
[ RUN      ] VkLayerTest.QueueFamilyMemoryScope
C:\j\w\32-bit\Release\Vulkan-ValidationLayers\tests\vkrenderframework.cpp(196): error: Failed
Did not receive expected error 'VUID-RuntimeSpirv-vulkanMemoryModel-06266'
```
Looks like `OpcodeMemoryScopePosition` might not be returning the expected result (or perhaps the test is wrong?)?

File: scripts/spirv_grammar_generator.py (hunk 1 above)
Comment (sfricke-samsung) [reply scripts/spirv_grammar_generator.py:292]: I have no argument against improving the styling, don't assume I am a Python-string-format-syntax guru or something

**Comment by sfricke-samsung** [comment]
> Looks like OpcodeMemoryScopePosition might not be returning the expected result (or perhaps the test is wrong?)?

I mixed up `MemoryScope` and `ExecutionScope` positions in the generated code, sorry about that

**Pull request merged**